extra/Lamdera/CLI/Format.hs

```diff
@@ -0,0 +1,245 @@
+
+elmFileOrDir :: Parser FilePath
+elmFileOrDir =
+  Parser
+    { _singular = "elm file or directory"
+    , _plural = "elm files or directories"
+    , _parser = Just  -- Accept any path
+    , _suggest = \_ -> return []
+    , _examples = \_ -> return ["Main.elm", "src/Main.elm"]
+    }
+
+command :: Terminal.Command
+command =
+  let
+    summary =
+      "Format Elm source files."
+
+    details =
+      "Usage: lamdera format [INPUT] [--output FILE] [--yes] [--validate] [--stdin]\n\n" ++
+      "  Format Elm source files."
+
+    example =
+      stack
+        [ reflow "Examples:"
+        , P.vcat [ P.indent 2 $ P.green "lamdera format Main.elm                     # formats Main.elm"
+                 , P.indent 2 $ P.green "lamdera format Main.elm --output Main2.elm  # formats Main.elm as Main2.elm"
+                 , P.indent 2 $ P.green "lamdera format src/                         # format all *.elm files in the src directory"
+                 ]
+        , reflow "Full guide to using elm-format at <https://github.com/avh4/elm-format>"
+        ]
+
+    formatFlags =
+      flags Format
+        |-- onOff "yes" "Reply 'yes' to all automated prompts."
+        |-- onOff "validate" "Check if files are formatted without changing them."
+        |-- onOff "stdin" "Read from stdin, output to stdout."
+        |-- flag "output" output_ "Write output to FILE instead of overwriting the given source file."
+        |-- onOff "help" "Show this help text."
+  in
+  Terminal.Command "format" (Common summary) details example (zeroOrMore elmFileOrDir) formatFlags run
+
+run :: [FilePath] -> Format -> IO ()
+run inputs flags =
+  if _help flags || null inputs && not (_stdin flags)
+    then do
+      TIO.putStrLn "Usage: lamdera format [INPUT] [--output FILE] [--yes] [--validate] [--stdin]\n"
+      TIO.putStrLn "  Format Elm source files.\n"
+      TIO.putStrLn "Available options:"
+      TIO.putStrLn "  --help                Show this help text"
+      TIO.putStrLn "  --output FILE         Write output to FILE instead of overwriting the given"
+      TIO.putStrLn "                        source file."
+      TIO.putStrLn "  --yes                 Reply 'yes' to all automated prompts."
+      TIO.putStrLn "  --validate            Check if files are formatted without changing them."
+      TIO.putStrLn "  --stdin               Read from stdin, output to stdout.\n"
+      TIO.putStrLn "Note: All flags must use double dashes (--flag). Single dashes (-flag) are not supported.\n"
+      TIO.putStrLn "Examples:"
+      TIO.putStrLn "  lamdera format Main.elm                     # formats Main.elm"
+      TIO.putStrLn "  lamdera format Main.elm --output Main2.elm  # formats Main.elm as Main2.elm"
+      TIO.putStrLn "  lamdera format src/                         # format all *.elm files in the src directory\n"
+      TIO.putStrLn "Full guide to using elm-format at <https://github.com/avh4/elm-format>"
+      Exit.exitSuccess
+    else case inputs of
```

Comment:
Wondering if it was possible to patch directly into elm-format's existing handling code here rather than replicating it? Or was there some problem doing so?